### PR TITLE
Two Ports Solution

### LINKED ISSUE #51: Two different PORT approach
#50 
The main idea is to give pvm.py a port args(which is already there).
So in the `etc/xdg/lxsession/LXDE-pi/autostart`
```
@lxterminal -e sh $HOME/PVM/launch 8001
@lxterminal -e sh $HOME/PVM/launch 8002
```
Ang in the `max-init.txt` will be 
```
1, jellyfish720.mp4 192.168.0.235 8001;
2, jellyfish720.mp4 192.168.0.235 8002;
```
And if we run pvm.py without port arg the default port is 8001.

### PR DESCRIPTION
**Pull request by KaneBetter**
Fixes #51 

Changed autostart section.
Changed log section.
Changed run section.
Rewrite pvm.py to accommodate different variations of PORT.
Update build_omxplayer.sh with instructions and folder check.

In this version, we can control two displays independently.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -100,7 +100,14 @@ On each Pi device, start the script
...
 ```bash
 cd PVM
+# Activate python env
+source PVM/bin/activate
+# The default port is 8001
 python pvm.py
+# If you want a different port, example 8002
+python pvm.py --port 8002
+# Or with launsh script
+sh launsh.sh 8001
```
Comment (omarcostahamido) [inline README.md]:
```suggestion
sh launch.sh 8001
```

File: pvm.py [hunk 2]
```diff
@@ -9,84 +10,88 @@
...
+		# Wait until current time is equal to next_time
+		while True:
+			now = datetime.now()
+			now = now.replace(microsecond=0)
+			if now.time() >= next_time.time():
+				break
+			sleep(0.005)
+
+	# Get value
+	if len(args) >= 6:
```
Comment (omarcostahamido) [inline pvm.py:66]: ok good. i like this. As in, it will also allow for us to think of future commands with multiple values

File: pvm.py [hunk 3]
```diff
@@ -9,84 +10,88 @@
...
 		value = args[5]
 
+	# If PORT == 8001, we export our video in the main display 2,
+	# else we export our video on the extra display 7
+	params = ['--loop', '--display']
+	if PORT == 8001:
+		params.append('2')
```
Comment (omarcostahamido) [inline pvm.py:73]: this needs to be explained in the readme

File: pvm.py [hunk 4]
```diff
@@ -9,84 +10,88 @@
...
+    		# If the file is already set, we exit OMX first
+			if IS_FILE_SET:
+				OMX.quit()
+				_logger.info("Quit previous file: %s", VIDEO_PATH)
+				IS_FILE_SET = False
+			VIDEO_PATH = PREFIX_VIDEOS_PATH + value
+			OMX = OMXPlayer(VIDEO_PATH, dbus_name='org.mpris.MediaPlayer2.omxplayer1', args=params)
+			# Because OMX has no user interface and will play automatically once the OMXPlayer object is created, 
+			# we need to call the puase() method after it is created.
+			OMX.pause()
```
Comment (omarcostahamido) [inline pvm.py:93]: ok thx

File: pvm.py [hunk 5]
```diff
@@ -95,68 +100,98 @@ def parse_commands(*args):
...
-				_logger.info("%s command failed.", command)
+			fps = get_rate(value)
+			if IS_FILE_SET:
+				OMX.quit()
+			params = params + ['--force-fps', fps]
+			OMX = OMXPlayer(VIDEO_PATH, dbus_name='org.mpris.MediaPlayer2.omxplayer1', args=params)
+			OMX.pause()
+			CAN_START = True
+			CAN_PAUSE = False
+			_logger.info("%s command success set rate: %s. You can start now!", command, fps)
```
Comment (omarcostahamido) [inline pvm.py]: oh, so changing the rate, requires reloading the file? that is really bad news.
Our goal was to change the rate interactively as the video is playing... make it go faster or slower.
I will be really surprised if that is not possible to achieve, it is a very simple effect, right?

**Comment by omarcostahamido** [comment]
I will leave it to the team to decide when to approve the PR. This sounds like an alternative to #31  which already has some work into it as well. I guess you will finish these up and try them?

File: README.md (hunk 1 above)
Comment (KaneBetter) [reply README.md]: Done

File: pvm.py (hunk 3 above)
Comment (KaneBetter) [reply pvm.py:73]: Done

File: pvm.py (hunk 5 above)
Comment (KaneBetter) [reply pvm.py]: Done, please see the commit [here](https://github.com/KaneBetter/omxplayer/commit/9474c053dfcbe6dde6da34593880334f4d1566f9).

**Review by SSSSam141 — APPROVED** [review]
LGTM

File: README.md [hunk 6]
```diff
@@ -116,10 +123,13 @@ optional arguments:
                  Default port is 8001
 ```
 
-On the control machine first edit the [max-init.txt](https://github.com/omarcostahamido/PVM/blob/main/max-init.txt) file. For each Pi device add a numbered line with the video filename, ip, and port. As in:
+On the control machine first edit the [max-init.txt](https://github.com/omarcostahamido/PVM/blob/main/max-init.txt) file. For each Pi device add a numbered line with the video filename, ip, and port. 
+By default, we use port 8001 to receive UDP for our main display. 8002 is used for additional displays. 
+As in:
```
Comment (omarcostahamido) [inline README.md:128]: It is not only the default port, it is also mandatory to leave it as 8001 for the first display, if the user wants to use 2 displays on the same device.
Also: not 8002 but any other port

Comment (KaneBetter) [reply README.md:128]: Done here. https://github.com/omarcostahamido/PVM/commit/a8fb622bbfde7faf534e81d919ba87bc98131cb1

**Pull request merged**